src/ol_infrastructure/components/aws/database.py

```diff
@@ -9,30 +9,64 @@
 - Create relevant security groups
 - Create DB instance
 """
-from typing import Text, Dict
+from enum import Enum
+from typing import Dict, List, Optional, Text, Union
 
 import pulumi
 from pulumi_aws import rds
-from pydantic import validator
+from pydantic import BaseModel, PositiveInt, SecretStr, conint, validator
 
+from ol_infrastructure.lib.aws.rds_helper import (
+    db_engines,
+    parameter_group_family
+)
 from ol_infrastructure.lib.ol_types import AWSBase
-from ol_infrastructure.lib.aws.rds_helper import db_engines, parameter_group_family
 
-# ensure db subnet group
-# create parameter group
 # manage backup policy
 # generate or retrieve password in config
 # create DB security group
 # storage encrypted
 
 
+class StorageType(str, Enum):
+    magnetic = 'standard'
+    ssd = 'gp2'
+    performance = 'io1'
+
+
+class OLReplicaDBConfig(BaseModel):
+    instance_size: Text = 'db.t3.small'
+    storage_type: StorageType = StorageType.ssd
+    public_access: bool = False
+    security_groups: Optional[List[rds.SecurityGroup]] = None
+
+    class Config:
+        arbitrary_types_allowed = True
+
+
 class OLDBConfig(AWSBase):
-    db_name: Text
     engine: Text
-    target_vpc_id: Text
     engine_version: Text
+    instance_name: Text  # The name of the RDS instance
+    password: SecretStr
+    subnet_group_name: Union[Text, pulumi.Output[str]]
+    security_groups: List[rds.SecurityGroup]
+    backup_days: conint(ge=0, le=35, strict=True) = 30
+    db_name: Optional[Text] = None  # The name of the database schema to create
+    instance_size: Text = 'db.m5.large'
+    is_public: bool = False
+    max_storage: Optional[PositiveInt] = None  # Set to allow for storage autoscaling
     multi_az: bool = True
+    prevent_delete: bool = True
     public_access: bool = False
```

Comment:
This depends on what the db will be used by, no?

Reply:
Yes, but it can be overridden. This is just a default.